### ISSUE TITLE
New 5-block networks trained from c83 for testing

### ISSUE THREAD

**Issue by bood** [leela-zero/leela-zero#862]
As mentioned in #167 , I trained some networks from self-plays of c83, to see any further improves can be made after we fixed the training script.

My settings below:
BATCH_SIZE = ~64~ 128 (I'm on a memory limited machine)
Learning rate = 0.00025 (~256/64=4~ 512/128=4, so learning rate is half of 0.0005 here)

@gcp Could you queue them on the server to test against c83? I'll continue uploading the weight files as the training goes on.

[leelaz-model-32000.txt.gz](https://github.com/gcp/leela-zero/files/1734332/leelaz-model-32000.txt.gz)
[leelaz-model-64000.txt.gz](https://github.com/gcp/leela-zero/files/1734333/leelaz-model-64000.txt.gz)
[leelaz-model-128000.txt.gz](https://github.com/gcp/leela-zero/files/1736869/leelaz-model-128000.txt.gz)
[leelaz-model-256000.txt.gz](https://github.com/gcp/leela-zero/files/1736870/leelaz-model-256000.txt.gz)
[leelaz-model-384000.txt.gz](https://github.com/gcp/leela-zero/files/1740652/leelaz-model-384000.txt.gz)
[leelaz-model-512000.txt.gz](https://github.com/gcp/leela-zero/files/1740653/leelaz-model-512000.txt.gz)
```
sha256sum:

c55b378d94688ea3d5ef3d89b70b6760fb93d28de540125bc85a4385f8cf7e94  leelaz-model-32000.txt
ae5792e3f25b155c63b6c378b70ab99157dc5ae7e3d00b2d0c3f7bf0382df6cf  leelaz-model-64000.txt
652da7aedb469133a50aaee09df6d4ca696a49051bd3ed4c682af3cc01c53c3b  leelaz-model-128000.txt
0e28d5fad254319cb9b8e5adefc3416d0bb577b1d354e3ebe755d379c1669c58  leelaz-model-256000.txt
```

```
step 32000, policy=2.66468 mse=0.180766 reg=0.0719167 total=3.45966 (395.04 pos/s)
step 32000, policy=2.67191 training accuracy=47.2207%, mse=0.190108
step 64000, policy=2.66586 mse=0.178327 reg=0.0721985 total=3.45137 (380.462 pos/s)
step 64000, policy=2.66133 training accuracy=47.4883%, mse=0.193403
step 128000, policy=2.66259 mse=0.180803 reg=0.0726888 total=3.45849 (505.743 pos/s)
step 128000, policy=2.66125 training accuracy=48.0664%, mse=0.200405
step 256000, policy=2.64801 mse=0.178686 reg=0.073512 total=3.43627 (367.615 pos/s)
step 256000, policy=2.65965 training accuracy=47.3496%, mse=0.186213
step 512000, policy=2.65111 mse=0.177092 reg=0.0747941 total=3.43427 (228.958 pos/s)
step 512000, policy=2.63303 training accuracy=48.4756%, mse=0.175107
```

(EDIT: Updated to use 128 batch size with 0.00025 learning rate, 32k and 64k networks uploaded)
(EDIT: 128k and 256k networks uploaded)

**Comment by gcp** [comment]
I don't think there's any point in testing a network from a 2.5% training accuracy.

**Comment by bood** [comment]
okay, I'm new to this so still learning.
Is this because the batch-size change? so the accuracy needs more steps to be reasonable? (I saw something like 50% from your previous posts) I do see it's raising though, about 8.8% at 88k steps. Any advise?

**Comment by Ttl** [comment]
Did you start from random weights? You should raise the learning rate at the beginning to speed it up, drop it as it stalls and run way more steps (>1M).

Starting from existing weights with net_to_model.py would probably be a better choice. Even that will require many steps because difference between network output and training data is big.

Your PC is so slow that fully training a network to convergence requires you several days of continuous training.

**Comment by gcp** [comment]
>Is this because the batch-size change? so the accuracy needs more steps to be reasonable?

What Ttl said basically. If you are starting from scratch you need a ton of training before it will be good.

**Comment by bood** [comment]
I see, I should start from c83 I guess. Let me try that.

**Comment by MingWR** [comment]
How about using 6b's selfplay games to train 5b network?

**Comment by bood** [comment]
@gcp I re-trained from c83 instead from random, updated the weights in the post now. `accuracy` seems to be much better now.

**Comment by zediir** [comment]
gcp is training with 512 batch size so learning rate should be eight and you need 64000 steps to equal the original 8000 steps

**Comment by bood** [comment]
@zediir Ouch, that's why I think the training parameters should be documented some where explicitly. It's probably true for the step numbers. I'll have to restart the training if the learning rate is incorrect though.

**Comment by gcp** [comment]
https://github.com/gcp/leela-zero/blob/36711d965d53cdaf65792ce79c5f1425f6abf1ed/training/tf/parse.py#L36

**Comment by bood** [comment]
Alright, my bad. Training restarted now, I increased the batch size to 128 so learning rate remains to be 0.00025.

**Comment by bood** [comment]
@gcp Updated with two initial trained networks (32k and 64k, which I believe corresponding to 4k and 16k with 512 batch size) with adjusted training parameters (128 and 0.00025). I'll continue uploading as training processes.

**Comment by sethtroisi** [comment]
closing, no active discussion for >1 year with many subsequent strong models
